compiler/rustc_ast/src/ast.rs

```diff
@@ -2355,6 +2355,19 @@ pub enum SelfKind {
     Explicit(P<Ty>, Mutability),
 }
 
+impl SelfKind {
+    pub fn as_suggestion(&self) -> String {
+        match self {
+            SelfKind::Value(mutbl) => mutbl.prefix_str().to_string(),
+            SelfKind::Region(None, mutbl) => mutbl.ref_prefix_str().to_string(),
+            SelfKind::Region(Some(lt), mutbl) => format!("&{lt} {}", mutbl.prefix_str()),
+            SelfKind::Explicit(_, _) => {
+                unreachable!("if we had an explicit self, we wouldn't be here")
```

Comment:
Makes sense to mark this as unreachable but I think the method name is a bit too nondescript, ppl might accidentally reach for it not knowing the internal invariants. Maybe the method name could be made more specific? 🤔